File: app/src/main/kotlin/com/culturesync/app/models/Technique.kt
package com.culturesync.app.models

data class Technique(
    val id: String,
    val name: String,
    val description: String,
    val difficulty: String,          // "Beginner", "Intermediate", "Advanced"
    val durationMinutes: Int,
    val thumbnailResId: Int = 0,
    val videoFileName: String? = null,
    val videoUrl: String? = null,    // dynamically fetched streaming url
    val hasTemplate: Boolean = true,
    val steps: List<String> = emptyList()
)

File: culturesync-backend/src/models/Technique.js
const mongoose = require('mongoose');

const techniqueSchema = new mongoose.Schema(
  {
    techniqueId:    { type: String, required: true, unique: true },
    name:           { type: String, required: true },
    description:    { type: String, required: true },
    difficulty:     { type: String, enum: ['Beginner', 'Intermediate', 'Advanced'], required: true },
    durationMinutes:{ type: Number, required: true },
    steps:          [{ type: String }],
    videoUrl:       { type: String, default: null },
    thumbnailUrl:   { type: String, default: null },
    isPublished:    { type: Boolean, default: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Technique', techniqueSchema);
